tooling/src/tasks/distribute.ts

```diff
@@ -0,0 +1,122 @@
+import { task, types } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { loadSchedule } from "../persistence";
+
+import {
+  calculateAmountToBridge,
+  createDistributeTxGC,
+  createDistributeTxMainnet,
+} from "../domain/distribution";
+
+import {
+  getAddressConfig,
+  getSafes,
+  SAFE_ADDRESS_GC,
+  SAFE_ADDRESS_MAINNET,
+  VESTING_ID,
+} from "../config";
+import Safe from "@gnosis.pm/safe-core-sdk";
+import SafeServiceClient from "@gnosis.pm/safe-service-client";
+import { SafeTransaction } from "@gnosis.pm/safe-core-sdk-types";
+import { Signer } from "ethers";
+
+task("distribute", "")
+  .addOptionalParam(
+    "safeMainnet",
+    "Safe Address in Mainnet",
+    SAFE_ADDRESS_MAINNET,
+    types.string,
+  )
+  .addOptionalParam(
+    "safeGC",
+    "Safe Address in Gnosis Chain",
+    SAFE_ADDRESS_GC,
+    types.string,
+  )
+  .addOptionalParam(
+    "vestingId",
+    "The vestingId on the VestingPool contract",
+    VESTING_ID,
+    types.string,
+  )
+  .setAction(async (taskArgs: any, hre: HardhatRuntimeEnvironment) => {
+    await hre.run("schedule:validate");
+
+    await hre.run("schedule:expand");
+
+    await hre.run("allocation:write-all");
+
+    const [merkleRootMainnet, merkleRootGC] = await hre.run(
+      "checkpoint:generate",
+    );
+
+    // move amount to bridge to checkpoint?
+    const { amountToClaim, amountToBridge } = await calculateAmountToBridge(
+      loadSchedule(),
+      hre,
+    );
+
+    const {
+      safeToken: safeTokenAddress,
+      vestingPool: vestingPoolAddress,
+      omniMediatorMainnet: omniMediatorAddress,
+      distroMainnet: distroMainnetAddress,
+      distroGC: distroGCAddress,
+    } = await getAddressConfig(hre);
+
+    const {
+      delegate,
+      safeSdkMainnet,
+      safeSdkGC,
+      serviceClientMainnet,
+      serviceClientGC,
+    } = await getSafes(taskArgs.safeMainnet, taskArgs.safeGC, hre);
+
+    const txMainnet = await createDistributeTxMainnet(safeSdkMainnet, {
+      safeTokenAddress,
+      vestingPoolAddress,
+      omniMediatorAddress,
+      distroMainnetAddress,
+      distroGCAddress,
+      vestingId: taskArgs.vestingId,
+      amountToClaim,
+      amountToBridge,
+      nextMerkleRoot: merkleRootMainnet,
+    });
+
+    const txGC = await createDistributeTxGC(safeSdkGC, {
+      distroAddress: distroGCAddress,
+      nextMerkleRoot: merkleRootGC,
+    });
+
+    await sanityCheck();
+
+    // Post To Safes
+    await propose(safeSdkMainnet, serviceClientMainnet, delegate, txMainnet);
+    await propose(safeSdkGC, serviceClientGC, delegate, txGC);
+  });
+
+async function sanityCheck() {
+  // check MerkleDistros are deployed
```

Comment:
this looks like it's gonna get fixed in your polish phase, but wanted to note it to make sure

Reply:
Since right now we don't have distribute task running, it's for the next stage. Right now we focus on getting harvest rolling and committing to master